include/copc-lib/io/writer.hpp

```diff
@@ -88,6 +90,12 @@ class Writer : public BaseIO
     // Adds a subpage to a given page
     Page AddSubPage(Page &page, VoxelKey key);
 
+    ~Writer()
+    {
+        if (writer_->IsOpen())
```

Comment:
virtual ~Writer() {}

Then, for the subclass:

~NamedWriter() override {
if (internal.isOpen()) internal.close();
named_stream.close();
delete named_stream;
}

This ensures the stream is not deleted before the file is closed, and also that if the NamedWriter gets downcast to Writer that the NamedWriter destructor still gets called.